socorro/processor/processor_app.py

```diff
@@ -188,63 +197,67 @@ def process_crash(self, crash_id, ruleset_name):
         ``destination``.
 
         """
+        self.logger.info("starting %s", crash_id)
         with sentry_sdk.push_scope() as scope:
             scope.set_extra("crash_id", crash_id)
             scope.set_extra("ruleset", ruleset_name)
 
-            # Fetch the raw crash data
-            try:
-                raw_crash = self.source.get_raw_crash(crash_id)
-                dumps = self.source.get_dumps_as_files(crash_id)
-            except CrashIDNotFound:
-                # If the crash isn't found, we just reject it--no need to capture
-                # errors here
-                self.processor.reject_raw_crash(
-                    crash_id, "crash cannot be found in raw crash storage"
-                )
-                return
-            except Exception as exc:
-                sentry_sdk.capture_exception(exc)
-                self.logger.exception("error: crash id %s: %r", crash_id, exc)
-                self.processor.reject_raw_crash(crash_id, f"error in loading: {exc}")
-                return
-
-            # Fetch processed crash data--there won't be any if this crash hasn't
-            # been processed, yet
-            try:
-                processed_crash = self.source.get_processed(crash_id)
-                new_crash = False
-            except CrashIDNotFound:
-                new_crash = True
-                processed_crash = {}
-
-            # Process the crash and remove any temporary artifacts from disk
-            try:
-                # Process the crash to generate a processed crash
-                processed_crash = self.processor.process_crash(
-                    ruleset_name, raw_crash, dumps, processed_crash
-                )
-
-                self.destination.save_processed_crash(raw_crash, processed_crash)
-                self.logger.info("saved - %s", crash_id)
-            except PolyStorageError as poly_storage_error:
-                # Capture and log the exceptions raised by storage backends
-                for storage_error in poly_storage_error:
-                    sentry_sdk.capture_exception(storage_error)
-                    self.logger.error("error: crash id %s: %r", crash_id, storage_error)
-                self.logger.warning("error in processing or saving crash %s", crash_id)
-
-                # Re-raise the original exception with the correct traceback
-                raise
-
-            finally:
-                # Clean up any dump files saved to the file system
-                for a_dump_pathname in dumps.values():
-                    if "TEMPORARY" in a_dump_pathname:
-                        try:
-                            os.unlink(a_dump_pathname)
-                        except OSError as x:
-                            self.logger.info("deletion of dump failed: %s", x)
+            with tempfile.TemporaryDirectory(dir=self.temporary_path) as tmpdir:
```

Comment:
When the context exits or the object is garbage collected, it will delete the contents of the temporary directory and the directory itself. That's a lot nicer than what we had before where we assumed the files created during processing were all dump files and in the `dumps` values.